src/NumericalAlgorithms/FiniteDifference/Reconstruct.hpp

```diff
@@ -24,34 +24,68 @@ namespace fd {
  * \brief Variable and flux vector splitting reconstruction schemes for finite
  * difference methods.
  *
- * Implementations of reconstruction methods must call this function
- * (`fd::reconstruction::detail::reconstruct`) to perform the reconstruction.
+ * Implementations of reconstruction methods must call the function
+ * `fd::reconstruction::detail::reconstruct` to perform the reconstruction.
  * This function performs the actual loops taking into account strides and ghost
- * zones for the reconstruction method. The type of reconstruction done is
- * specified with the `Reconstructor` explicit template parameter. Parameters
- * for the reconstruction scheme are forwarded along using the
- * `args_for_reconstruction` parameter pack.
+ * zones for the reconstruction method. The `reconstruct` function has the
+ * following signature:
+ *
+ * \code
+ * template <typename Reconstructor, typename... ArgsForReconstructor, size_t
+ *           Dim>
+ * void reconstruct(
+ *     const gsl::not_null<std::array<gsl::span<double>, Dim>*>
+ *         reconstructed_upper_side_of_face_vars,
+ *     const gsl::not_null<std::array<gsl::span<double>, Dim>*>
+ *         reconstructed_lower_side_of_face_vars,
+ *     const gsl::span<const double>& volume_vars,
+ *     const DirectionMap<Dim, gsl::span<const double>>& ghost_cell_vars,
+ *     const Index<Dim>& volume_extents, const size_t number_of_variables,
+ *     const ArgsForReconstructor&... args_for_reconstructor) noexcept;
+ * \endcode
```

Comment:
Do we have the option of using a doxygen snippet here? If so, that would have the advantage of avoiding the code and dox getting out of sync...

Reply:
I think so far we've been avoiding snippet comments in the source tree, but I don't know if that changed at some point.